pkg/cmd/marathon/app/app_start_test.go

```diff
@@ -0,0 +1,111 @@
+		case http.MethodPut:
+			w.Write([]byte(`{"deploymentId":"5ed4c0c5-9ff8-4a6f-a0cd-f57f59a34b43","version":"2015-09-29T15:59:51.164Z"}`))
+		}
+	}))
+	defer ts.Close()
+
+	out := new(bytes.Buffer)
+	env := mock.NewEnvironment()
+	env.Out = out
+	ctx := mock.NewContext(env)
+	cluster := config.NewCluster(nil)
+	cluster.SetURL(ts.URL)
+	ctx.SetCluster(cluster)
+
+	client, err := marathon.NewClient(ctx)
+	assert.NoError(t, err)
+
+	deployment, err := marathonAppStart(*client, "app-test", 1, false)
+	assert.Equal(t, "5ed4c0c5-9ff8-4a6f-a0cd-f57f59a34b43", deployment.DeploymentID)
+	assert.NoError(t, err)
+}
+
+func TestMarathonStartNotEnoughInstances(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	}))
+	defer ts.Close()
+
+	out := new(bytes.Buffer)
+	env := mock.NewEnvironment()
+	env.Out = out
+	ctx := mock.NewContext(env)
+	cluster := config.NewCluster(nil)
+	cluster.SetURL(ts.URL)
+	ctx.SetCluster(cluster)
+
+	client, err := marathon.NewClient(ctx)
+	assert.NoError(t, err)
+
+	_, err = marathonAppStart(*client, "app-test", 0, false)
+	assert.EqualError(t, err, "the number of instances must be positive: 0")
+}
+
+func TestMarathonStartTooManyInstances(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "/service/marathon/v2/apps/app-test", r.URL.String())
+		w.Write([]byte(`{"app":{"id":"/app-test","backoffFactor":1.15,"backoffSeconds":1,"cpus":1.0,"disk":64.0,"executor":"","instances":3}}`))
+	}))
+	defer ts.Close()
+
+	out := new(bytes.Buffer)
+	env := mock.NewEnvironment()
+	env.Out = out
+	ctx := mock.NewContext(env)
+	cluster := config.NewCluster(nil)
+	cluster.SetURL(ts.URL)
+	ctx.SetCluster(cluster)
+
+	client, err := marathon.NewClient(ctx)
+	assert.NoError(t, err)
+
+	_, err = marathonAppStart(*client, "app-test", 1, false)
+	assert.EqualError(t, err, "application 'app-test' already started: 3 instances")
+}
+
+func TestMarathonStartError(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "/service/marathon/v2/apps/app-test", r.URL.String())
+		switch method := r.Method; method {
+		case http.MethodGet:
+			w.Write([]byte(`{"app":{"id":"/app-test","backoffFactor":1.15,"backoffSeconds":1,"cpus":1.0,"disk":64.0,"executor":"","instances":0}}`))
+		case http.MethodPut:
+			w.WriteHeader(http.StatusNotFound)
+			w.Write([]byte("404 - Something bad happened!"))
+		}
+	}))
+	defer ts.Close()
+
+	out := new(bytes.Buffer)
+	env := mock.NewEnvironment()
+	env.Out = out
+	ctx := mock.NewContext(env)
+	cluster := config.NewCluster(nil)
+	cluster.SetURL(ts.URL)
+	ctx.SetCluster(cluster)
+
+	client, err := marathon.NewClient(ctx)
+	assert.NoError(t, err)
+
+	_, err = marathonAppStart(*client, "app-test", 1, false)
+	assert.EqualError(t, err, "Marathon API error: 404 - Something bad happened!")
```

Comment:
In the commands I've ported, I've tried to keep the error messages close to the same, though I've changed them to meet Go guidelines (no leading uppercase, no period). I'm not too committed to that method but we should pick a consistent way to dealing with errors from `go-marathon`. https://github.com/dcos/dcos-core-cli/pull/411/files#diff-4112de97288590870c674ea78812c862R51-R56

Reply:
In the previous commands I've ported I've done the same as you for the most common errors (e.g. 404, here the first 404 has special handling) while letting go-marathon do its things with the others. It seems similar in the snippet you sent.